### PR TITLE
Automate even more of the rewriter

### PR DESCRIPTION
**Pull request by JasonGross**
On top of #541

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +195,94 @@ The files contain:
...
+  taking in wf-goodness of rewrite rules as a hypothesis.
+
+- RewriterInterpProofs1.v: Proves interp-correctness of the generic
+  rewriter, taking in interp-goodness of rewrite rules as a
+  hypothesis.
+
+- RewriterProofsTactics.v: Defines the tactic
+  `RewriteRules.Tactic.make_rewriter` (and a similar tactic notation)
+  which build the entire `VerifiedRewriter`.  They take in the
+  `include_interp` as in Rewriter.v tactics, as well as an hlist of
```
Comment (andres-erbsen) [inline README.md]: could this list of proofs just be `fold_right conj . map fst`?

File: README.md [hunk 2]
```diff
@@ -194,6 +195,94 @@ The files contain:
...
+  rewrite rule types.  This is the primary interface for defining a
+  rewriter from a list of rewrite rules.
+
+- Rewriter/{NBE, Arith, ArithWithCasts, StripLiteralCasts, ToFancy,
+  ToFancyWithCasts}.v: Use the tactic from RewriterProofsTactics.v
+  together with the proven list of rewrite rules from
+  RewriterRulesProofs.v to reify and reduce the corresponding pass and
+  generate a rewriter.
+
+- RewriterProofs.v: `Definition`less file that `Export`s the rewriters
```
Comment (andres-erbsen) [inline README.md]: if it exports stuff other than proofs, perhaps `RewriterAll.v`?

File: README.md (hunk 1 above)
Comment (JasonGross) [reply README.md]: Do you mean `fold_right and True . map snd`?  (That is effectively what `hlist` does, except it uses `prod`.  Is there a reason to prefer the `fold` version over `hlist`?)

File: README.md (hunk 2 above)
Comment (JasonGross) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (andres-erbsen) [reply README.md]: Perhaps it doesn't matter indeed, just seeing "hlist" made me think of term encodings...

**Pull request merged**